packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -124,19 +123,20 @@ class SKPaymentQueueWrapper {
   ///
   /// The `applicationUserName` should match the original
   /// [SKPaymentWrapper.applicationUsername] used in [addPayment].
+  /// If no `applicationUserName` was used, `applicationUserName` should be null.
   ///
   /// This method either triggers [`-[SKPayment
   /// restoreCompletedTransactions]`](https://developer.apple.com/documentation/storekit/skpaymentqueue/1506123-restorecompletedtransactions?language=objc)
   /// or [`-[SKPayment restoreCompletedTransactionsWithApplicationUsername:]`](https://developer.apple.com/documentation/storekit/skpaymentqueue/1505992-restorecompletedtransactionswith?language=objc)
   /// depending on whether the `applicationUserName` is set.
-  Future<void> restoreTransactions({String applicationUserName}) async {
+  Future<void> restoreTransactions({String? applicationUserName}) async {
     await channel.invokeMethod<void>(
         '-[InAppPurchasePlugin restoreTransactions:result:]',
         applicationUserName);
   }
 
   // Triage a method channel call from the platform and triggers the correct observer method.
-  Future<dynamic> _handleObserverCallbacks(MethodCall call) {
+  Future<void> _handleObserverCallbacks(MethodCall call) async {
     assert(_observer != null,
         '[in_app_purchase]: (Fatal)The observer has not been set but we received a purchase transaction notification. Please ensure the observer has been set using `setTransactionObserver`. Make sure the observer is added right at the App Launch.');
```

Comment:
Consider setting a local `observer` variable so you're using `!` once where it's obvious why it's okay, rather than throughout  a somewhat long method.

Reply:
done!